meson.build: support generating version information from git

ejoerns opened the pull request:
This creates a `version.h` file with version information derived from `git describe` using `vcs_tag`.
The `version.h` include will be injected into rauc and librauc builds.

For `meson dist`, no git information will be available, thus `meson.project_version()` serves as a fallback.

However, the generated tarball then still produces a non-fatal message that might upset users:

> fatal: not a git repository (or any parent up to mount point /)

Note that meson does not seem to allow including generated files in dist tarballs by design, thus adding a '.tarball-version'-like file will not be possible.

<!--
Thank you for your pull request!

If this is your first pull request for RAUC, please read:
https://rauc.readthedocs.io/en/latest/contributing.html

Please describe what it changes, e.g. fixes a bug (and how), adds a feature, fixes documentation…

If you add a feature, please answer these questions:
- What do you use the feature for?
- How does RAUC benefit from the feature?
- How did you verify the feature works?
- If hardware is needed for the feature, which hardware is supported and which
  hardware did you test with?

Please also allow edits by maintainers, so we can apply minor fixes directly.
-->

<!--
In case your PR fixes an issue, please reference it in the next line, i.e.
Fixes: #[insert number without brackets here]
-->

eli-schwartz commented:
> However, the generated tarball then still produces a non-fatal message that might upset users:

This happens because you unconditionally run a `command: ['git', ...]`. If you don't specify one, meson's builtin command passed to the vcstagger handler script, is `git describe --dirty=+`, ***but*** if the current project isn't a git repository then it uses a command that fails silently.

> Note that meson does not seem to allow including generated files in dist tarballs by design, thus adding a '.tarball-version'-like file will not be possible.

You're supposed to use `meson.add_dist_script()` for this, to post-process the staging directory of the dist tarball before it is tarred up.

ejoerns commented:
> > However, the generated tarball then still produces a non-fatal message that might upset users:
> 
> This happens because you unconditionally run a `command: ['git', ...]`. If you don't specify one, meson's builtin command passed to the vcstagger handler script, is `git describe --dirty=+`, _**but**_ if the current project isn't a git repository then it uses a command that fails silently.

@eli-schwartz Thanks again for your valuable feedback! Somehow I wasn't aware that it works with a built-in command, too. And it also just seems to fall back to the meson project version if no git is available.

I've updated my PR to use this. What is open to discuss is if we need the git-generated version in the tarball, too or if the hard-coded value is sufficient for us.

> > Note that meson does not seem to allow including generated files in dist tarballs by design, thus adding a '.tarball-version'-like file will not be possible.
> 
> You're supposed to use `meson.add_dist_script()` for this, to post-process the staging directory of the dist tarball before it is tarred up.

Good hint, haven't found this in documentation / examples so far. I've added a minimal hack for supporting autotools in getting basic version information (meson project version first of all).
